hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -403,6 +407,47 @@ func (r *HostedClusterReconciler) reconcile(ctx context.Context, req ctrl.Reques
 			log.Info("hostedcluster is still deleting", "name", req.NamespacedName)
 			return ctrl.Result{RequeueAfter: clusterDeletionRequeueDuration}, nil
 		}
+
+		// This new condition is necessary for OCM personnel to report any cloud dangling objects to the user.
+		// The grace period is customizable using the flag --destroy-grace-period. It's a time.Duration field type.
+		// This condition only will be created when !SelfManage and CloudResourcesDestroyedCondition is False.
+		CloudResourcesDestroyedCondition := meta.FindStatusCondition(hcluster.Status.Conditions, string(hyperv1.CloudResourcesDestroyed))
+		// ToDo (jparrill): when you deploy/destroy the cluster form CLI, you need to known the HCDestroyGracePeriod if not, on destroy
+		// the timeout will avoid to terminate the cluster properly. So we need to add an annotation with the HCDestroyGracePeriod for each
+		// HC deployed when tha value is > 0.
+		if r.HCDestroyGracePeriod > 0 {
+			reportingGracePeriod := &metav1.Condition{
+				Type:   string(hyperv1.HostedClusterDestroyed),
+				Status: metav1.ConditionFalse,
+				Reason: hyperv1.WaitingForGracePeriodReason,
+			}
+
+			HostedClusterDestroyedCondition := meta.FindStatusCondition(hcluster.Status.Conditions, string(hyperv1.HostedClusterDestroyed))
+			if HostedClusterDestroyedCondition != nil {
+				reportingGracePeriod = HostedClusterDestroyedCondition
+			}
+
+			reportingGracePeriod.Message = fmt.Sprintf("Resources: %v", CloudResourcesDestroyedCondition.Message)
+
+			meta.SetStatusCondition(&hcluster.Status.Conditions, *reportingGracePeriod)
+			if err := r.Client.Status().Update(ctx, hcluster); err != nil {
+				return ctrl.Result{}, fmt.Errorf("failed to update status: %w", err)
+			}
+
+			// If now is after the condition creation date + grace period, update condition and continue, else requeue
+			if time.Since(reportingGracePeriod.LastTransitionTime.Time) > r.HCDestroyGracePeriod {
+				reportingGracePeriod.Status = metav1.ConditionTrue
+				reportingGracePeriod.Reason = hyperv1.AsExpectedReason
+			} else {
+				return ctrl.Result{RequeueAfter: ReportingGracePeriodRequeueDuration}, nil
+			}
+
+			meta.SetStatusCondition(&hcluster.Status.Conditions, *reportingGracePeriod)
+			if err := r.Client.Status().Update(ctx, hcluster); err != nil {
+				return ctrl.Result{}, fmt.Errorf("failed to update status: %w", err)
+			}
```

Comment:
Please remove, the HostedCluster will be deleted as soon as this function returns, no need to update its status.